### PR TITLE
fix plsql grammar bugs

### PR DESCRIPTION
**Pull request by hellozrh**
1. oracle 21c grammar: alter table with MEMOPTIMIZE clause.
2. add flashback_table.
3. fix index_org_table_clause.

### CODE REVIEW

**Comment by msagca** [comment]
@hellozrh,

Title is misleading. You've made changes to an already existing grammar.

File: sql/plsql/PlSqlLexer.g4 [hunk 1]
```diff
@@ -2225,6 +2226,7 @@ LAG:                          'LAG';
 LEAD:                         'LEAD';
 MAX:                          'MAX';
 MEDIAN:                       'MEDIAN';
+MEMOPTIMIZE:                  'MEMOPTIMIZE';
```
Comment (KvanTTT) [inline sql/plsql/PlSqlLexer.g4:2229]: I guess it also should be added to `non_reserved_keywords_in_12c` or to `non_reserved_keywords_pre12c`.

File: sql/plsql/PlSqlParser.g4 [hunk 2]
```diff
@@ -1272,6 +1274,19 @@ drop_index
     : DROP INDEX index_name ';'
     ;
 
+flashback_table
+    : FLASHBACK TABLE tableview_name (',' tableview_name)* TO
+      (
+        (( (SCN | TIMESTAMP) expression | RESTORE POINT restore_point ) ((ENABLE | DISABLE) TRIGGERS)? )
+        |
+        (BEFORE DROP (RENAME TO tableview_name)?)
+      )
```
Comment (KvanTTT) [inline sql/plsql/PlSqlParser.g4:1283]: Please fix whitespaces and parentheses:

```suggestion
      ( ((SCN | TIMESTAMP) expression | RESTORE POINT restore_point) ((ENABLE | DISABLE) TRIGGERS)?
        | BEFORE DROP (RENAME TO tableview_name)?
      )
```

File: sql/plsql/PlSqlParser.g4 [hunk 3]
```diff
@@ -6569,6 +6589,7 @@ non_reserved_keywords_pre12c
     | SWITCHOVER
     | SYNCHRONOUS
     | SYNC
+    | SYS
```
Comment (KvanTTT) [inline sql/plsql/PlSqlParser.g4:6592]: I guess it should be either in `non_reserved_keywords_in_12c` or in `non_reserved_keywords_pre12c` but not in both.

**Review by KvanTTT — CHANGES_REQUESTED** [review]

**Pull request merged**